src/hera/workflows/_mixins.py

```diff
@@ -579,34 +579,39 @@ def _build_arguments(self) -> Optional[ModelArguments]:
         elif isinstance(self.arguments, ModelArguments):
             return self.arguments
 
+        def add_argument(k: str, v: Any, result: ModelArguments):
+            if isinstance(v, Parameter):
+                value = v.with_name(k).as_argument()
+                result.parameters = [value] if result.parameters is None else result.parameters + [value]
```

Comment:
note: this can be simplified to
```suggestion
                result.parameters = (result.parameters or []) + [value]
```

Reply:
Very good suggestion 🚀